tests/flows/test_reuse_order_fees.py

```diff
@@ -0,0 +1,328 @@
+        publish_market_order_fee_address=publish_market_wallet.address,
+        publish_market_order_fee_token=bt.address,
+        publish_market_order_fee_amount=parse_units("10", bt.decimals()),
+        timeout=5,
+    )
+
+    # Mint 50 datatokens in consumer wallet from publisher.
+    dt.mint(
+        account_address=consumer_wallet.address,
+        value=to_wei("50"),
+        from_wallet=publisher_wallet,
+    )
+
+    # Mock non-zero provider fees (simulate first time paying provider fees)
+    provider_fee = parse_units(provider_fee_in_unit, bt.decimals())
+    valid_until = int((datetime.utcnow() + timedelta(seconds=10)).timestamp())
+    provider_fees = get_provider_fees(
+        web3,
+        provider_wallet,
+        bt.address,
+        provider_fee,
+        valid_until,
+    )
+
+    # Grant datatoken infinite approval to spend consumer's base tokens
+    bt.approve(dt.address, MAX_WEI, consumer_wallet)
+
+    # Start order: pay order fees and provider fees
+    start_order_tx_id = dt.start_order(
+        consumer=consumer_wallet.address,
+        service_index=asset.get_index_of_service(service),
+        provider_fee_address=provider_fees["providerFeeAddress"],
+        provider_fee_token=provider_fees["providerFeeToken"],
+        provider_fee_amount=provider_fees["providerFeeAmount"],
+        v=provider_fees["v"],
+        r=provider_fees["r"],
+        s=provider_fees["s"],
+        valid_until=provider_fees["validUntil"],
+        provider_data=provider_fees["providerData"],
+        consume_market_order_fee_address=consume_market_wallet.address,
+        consume_market_order_fee_token=bt.address,
+        consume_market_order_fee_amount=parse_units("10", bt.decimals()),
+        from_wallet=consumer_wallet,
+    )
+
+    # Reuse order where:
+    #     Order: valid
+    #     Provider fees: valid
+    # Simulate valid provider fees by setting them to 0
+    reuse_order_with_mock_provider_fees(
+        provider_fee_in_unit="0",
+        start_order_tx_id=start_order_tx_id,
+        bt=bt,
+        dt=dt,
+        publisher_wallet=publisher_wallet,
+        publish_market_wallet=publish_market_wallet,
+        consume_market_wallet=consume_market_wallet,
+        consumer_wallet=consumer_wallet,
+        provider_wallet=provider_wallet,
+        web3=web3,
+    )
+
+    # Reuse order where:
+    #     Order: valid
+    #     Provider fees: expired
+    # Simulate expired provider fees by setting them to non-zero
+    reuse_order_with_mock_provider_fees(
+        provider_fee_in_unit=provider_fee_in_unit,
+        start_order_tx_id=start_order_tx_id,
+        bt=bt,
+        dt=dt,
+        publisher_wallet=publisher_wallet,
+        publish_market_wallet=publish_market_wallet,
+        consume_market_wallet=consume_market_wallet,
+        consumer_wallet=consumer_wallet,
+        provider_wallet=provider_wallet,
+        web3=web3,
+    )
+
+    # Sleep for 6 seconds, long enough for order to expire
+    sleep(6)
```

Comment:
can we replace this with a loop? It would help with CI consistency.

Reply:
Okay sure.

Reply:
On second thought @calina-c, I'm struggling to find an easy way to check if an order is expired inside the loop.

So for now, I think it would be okay to merge this PR with `sleep(6)`, and replace it later.  I don't want to hold up this PR any longer.

I created an issue in the provider to improve the `initialize` endpoint so that it can check if an existing order is valid.
https://github.com/oceanprotocol/provider/issues/461